webapp/src/js/components/Map/TableMarkersLayer.js

```diff
@@ -282,6 +319,27 @@ let TableMarkersLayer = React.createClass({
       });
   },
 
+  composeLegendControl(payload) {
+    let {table, markerColourProperty} = payload;
+
+    let legendControl = null;
+
+    if (table !== undefined && markerColourProperty !== undefined) {
+      let legend = ReactDOMServer.renderToStaticMarkup(
```

Comment:
I see, you mean effectively return `{component: <PropertyLegend [...] />, [...]}` and then do something like `div.innerHTML = ReactDOMServer.renderToStaticMarkup(control.component)` in Map instead? OK, will do.